bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java

```diff
@@ -211,85 +208,89 @@ public void receivePercentageTest() throws InterruptedException, ExecutionExcept
 
         c.processMessage("state", "ON".getBytes()); // Normal on state
         assertThat(value.getChannelState().toString(), is("0,0,10"));
-        assertThat(value.getMQTTpublishValue().toString(), is("0,0,10"));
+        assertThat(value.getMQTTpublishValue(), is("0,0,10"));
 
         c.processMessage("state", "FOFF".getBytes()); // Custom off state
         assertThat(value.getChannelState().toString(), is("0,0,0"));
-        assertThat(value.getMQTTpublishValue().toString(), is("0,0,0"));
+        assertThat(value.getMQTTpublishValue(), is("0,0,0"));
 
         c.processMessage("state", "10".getBytes()); // Brightness only
         assertThat(value.getChannelState().toString(), is("0,0,10"));
-        assertThat(value.getMQTTpublishValue().toString(), is("0,0,10"));
+        assertThat(value.getMQTTpublishValue(), is("0,0,10"));
 
         c.processMessage("state", "12,18,100".getBytes());
         assertThat(value.getChannelState().toString(), is("12,18,100"));
-        assertThat(value.getMQTTpublishValue().toString(), is("12,18,100"));
+        assertThat(value.getMQTTpublishValue(), is("12,18,100"));
     }
 
     @Test
-    public void receiveLocationTest() throws InterruptedException, ExecutionException, TimeoutException {
+    public void receiveLocationTest() {
         LocationValue value = new LocationValue();
         ChannelState c = spy(new ChannelState(config, channelUID, value, channelStateUpdateListener));
         c.start(connection, mock(ScheduledExecutorService.class), 100);
 
         c.processMessage("state", "46.833974, 7.108433".getBytes());
         assertThat(value.getChannelState().toString(), is("46.833974,7.108433"));
-        assertThat(value.getMQTTpublishValue().toString(), is("46.833974,7.108433"));
+        assertThat(value.getMQTTpublishValue(), is("46.833974,7.108433"));
     }
 
     @Test
-    public void receiveDateTimeTest() throws InterruptedException, ExecutionException, TimeoutException {
+    public void receiveDateTimeTest() {
         DateTimeValue value = new DateTimeValue();
-        ChannelState c = spy(new ChannelState(config, channelUID, value, channelStateUpdateListener));
-        c.start(connection, mock(ScheduledExecutorService.class), 100);
+        ChannelState subject = spy(new ChannelState(config, channelUID, value, channelStateUpdateListener));
+        subject.start(connection, mock(ScheduledExecutorService.class), 100);
 
         ZonedDateTime zd = ZonedDateTime.now();
         String datetime = zd.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
-        c.processMessage("state", datetime.getBytes());
-        assertThat(value.getChannelState().toString(), is(datetime + "+0100"));
-        assertThat(value.getMQTTpublishValue().toString(), is(datetime));
+
+        subject.processMessage("state", datetime.getBytes());
+
+        String channelState = value.getChannelState().toString();
+        assertTrue("Expected '" + channelState + "' to start with '" + datetime + "'",
```

Comment:
I have been told in the Onewire binding (in ESH) not to use this pattern (additional message in asserts), Has this changed? I personally think that this makes a lot tests more clear.

Reply:
I guess there was a way to use `assertThat` in your case which is preferred I think. But I'm happy with `assertTrue` as well in this test.

Reply:
@j-n-k do you have a link to this I’m curious about the why?

Reply:
Unfortunately I can't find that comment anymore. It seems that it wasn't in the initial contribution but at a later point. Could also have been in the DMX binding. IIRC the argument was that the test itself is enough description and the additional string just pollutes the code.